habitat-lab/habitat/gym/gym_wrapper.py

```diff
@@ -16,23 +16,20 @@
 from habitat.core.spaces import EmptySpace
 from habitat.utils.visualizations.utils import observations_to_image
 
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+
+    from habitat.core.dataset import BaseEpisode
+    from habitat.core.env import RLEnv
+
 try:
     import pygame
 
 except ImportError:
     pygame = None
 
 
-def flatten_dict(d, parent_key=""):
-    # From https://stackoverflow.com/questions/6027558/flatten-nested-dictionaries-compressing-keys
-    items = []
-    for k, v in d.items():
-        new_key = parent_key + "." + str(k) if parent_key else str(k)
-        if isinstance(v, dict):
-            items.extend(flatten_dict(v, new_key).items())
-        else:
-            items.append((new_key, v))
-    return dict(items)
```

Comment:
I would rather we flatten the dictionary in the environment rather than in habitat-baselines. Is there a reason you are reverting this change?

Reply:
Please see the motivation I've added to the PR description.